File: src/org/cerion/symcalc/expression/function/arithmetic/Sqrt.java
package org.cerion.symcalc.expression.function.arithmetic;

import org.cerion.symcalc.exception.ValidationException;
import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.FunctionExpr;
import org.cerion.symcalc.expression.number.RationalNum;

public class Sqrt extends FunctionExpr {
    public Sqrt(Expr... e) {
        super(FunctionType.SQRT,e);
    }

    @Override
    protected Expr evaluate() {
        return new Power(get(0), new RationalNum(1,2)).eval();
    }

    @Override
    public void validate() throws ValidationException {
        validateParameterCount(1);
    }
}

File: src/org/cerion/symcalc/expression/function/arithmetic/Sqrt.kt
package org.cerion.symcalc.expression.function.arithmetic

import org.cerion.symcalc.exception.ValidationException
import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.expression.function.Function
import org.cerion.symcalc.expression.function.FunctionExpr
import org.cerion.symcalc.expression.number.RationalNum

class Sqrt(vararg e: Expr) : FunctionExpr(Function.SQRT, *e) {

    override fun evaluate(): Expr {
        return Power(get(0), RationalNum(1, 2)).eval()
    }

    @Throws(ValidationException::class)
    override fun validate() {
        validateParameterCount(1)
    }
}